numba/npyufunc/ufuncbuilder.py

```diff
@@ -66,6 +67,71 @@ def compile(self, sig, locals={}, **targetoptions):
 
 target_registry['npyufunc'] = UFuncDispatcher
 
+# Utility functions
+
+def _compile_element_wise_function(nb_func, targetoptions, sig=None,
+                                   argtypes=None, restype=None):
+    # Handle argtypes
+    if argtypes is not None:
+        warnings.warn("Keyword argument argtypes is deprecated",
+                      DeprecationWarning)
+        assert sig is None
+        if restype is None:
+            sig = tuple(argtypes)
+        else:
+            sig = restype(*argtypes)
+
+    # Do compilation
+    # Return CompileResult to test
+    cres = nb_func.compile(sig, **targetoptions)
+    args, return_type = sigutils.normalize_signature(sig)
+
+    return cres, args, return_type
+
+def _check_ufunc_signature(cres, args, return_type):
```

Comment:
I agree that the method name (`_finalize_signature()`) was more descriptive.  Will do.